### PR TITLE
Add and remove

### PR DESCRIPTION
**Pull request by romans-adi**
For this milestone:

- I used ES6 modules to restructure my project.
- All modules have descriptive names related to their functionality.
- To Do Model calls all functions as class methods. 
- All new tasks are not completed by default.
- Indexes updates on removing (map used).
- I used a textarea tag for description editing functionally. Since I prevent default behavior on the key 'Enter' press and save the changed value, there's no particular function for this created.

### Update for the second review

- A 'remove' button for each task added on textarea focus event.
- Loop added to reset all indexes. 
- Type in class fixed.
- Two functions for the next project were commented out (not removed). Functionality is not broken.

### Update for the third review

- Functionality for `clear all` and `checkbox` was removed.

_To run the project in dev, please use `npm start`_

### CODE REVIEW

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @romans-adi 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Removed all hardcoded items from the tasks array ✔️ 
Implement a function for adding, deleting and editing a new task ✔️ 
All changes to the To Do List saved in local storage ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/remove.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+function removeTask(index) {
+  const tasks = getTasks();
+  tasks.splice(index, 1);
+  this.tasks = this.tasks.map((task, i) => {
+    task.index = i + 1;
+    return task;
+  });
+  setTasks(tasks);
+  render(tasks);
+}
```
[the remark this answers is not in the document]
Comment (romans-adi) [reply src/modules/remove.js:13]: I agree that for better UI practice it is a good idea to add a remove button, but it was not a requirement.
Technically this logic was done with a clearButton and it was not forbidden.
Indexes still need to be fixed. Thank you!

File: src/modules/render.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  clearButton.classList.add('clear-btn');
+  clearButton.textContent = 'Clear completed tasks';
+  // clearButton.addEventListener('click', () => {
+  //   tasks = tasks.filter((task) => !task.completed);
+  //   tasks.forEach((task, index) => {
+  //     task.index = index + 1;
+  //   });
+  //   setTasks(tasks);
+  //   render(tasks);
+  // });
```
Comment (OyePriscilla) [inline src/modules/render.js:134]: - [ ] Kindly remove this commented lines of code so as to have a clean codebase. 👍 _**I observed that it was not commented out in the last review. 👍**_

File: src/modules/render.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    const checkbox = document.createElement('input');
+    checkbox.classList.add('checkbox');
+    checkbox.type = 'checkbox';
+    // checkbox.checked = task.completed;
+    // checkbox.addEventListener('change', (event) => {
+    //   task.completed = event.target.checked;
+    //   setTasks(tasks);
+    //   render(tasks);
+    // });
```
Comment (OyePriscilla) [inline src/modules/render.js:73]: - [ ] Kindly remove this commented lines of code so as to have a clean codebase. 👍 _**I observed that it was not commented out in the last review. 👍**_

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @romans-adi 👏 👏 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Removed all hardcoded items from the tasks array ✔️
Implement a function for adding, deleting and editing a new task ✔️
All changes to the To Do List saved in local storage ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by braincee — APPROVED** [review]
Hi @romans-adi 👋  , I am @braincee, your Code Reviewer.

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

youre-approved-shirley-auto-group

OPTIONAL CHANGES
---

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @braincee._

---

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**